### PR TITLE
Fail fast improvements

### PR DESCRIPTION
**Pull request by gregsexton**
A couple of improvements to get things to fail faster rather than allowing you to continue with bad assumptions.

- We check table, view, and schema creation to make sure when idempotent you really are getting the expected result. This prevents you moving forward with a misaligned schema.

- Made sure that the dev server checks the result of process calls and does not drop valuable error info.

### CODE REVIEW

File: sql/server/core/src/main/kotlin/core/Skdb.kt [hunk 1]
```diff
@@ -22,9 +22,16 @@ data class ProcessOutput(val output: ByteArray, val exitCode: Int) {
...
-  fun getOrThrow(): ByteArray {
+  fun logOnError(): ProcessOutput {
     if (exitCode != 0) {
       System.err.println(decode())
+    }
+    return this
+  }
+
+  fun getOrThrow(): ByteArray {
+    if (exitCode != 0) {
```
Comment (jberdine) [inline sql/server/core/src/main/kotlin/core/Skdb.kt]: nit: some code tests `exitCode` and some code calls `exitSuccessfully`, is there a pattern?

**Review by jberdine — COMMENTED** [review]
The code change LGTM, but some of the existing tests hit the new checks, and some others fail but I haven't investigated.

**Comment by gregsexton** [comment]
I'll investigate the test failures.

File: sql/server/core/src/main/kotlin/core/Skdb.kt (hunk 1 above)
Comment (gregsexton) [reply sql/server/core/src/main/kotlin/core/Skdb.kt]: exitSuccessfully is the interface. The implementation was just using the field (which admittedly isn't private so could be used externally). I change the implementation to call exitSuccessfully.

**Pull request merged**